src/Simulation/Simulators/SparseSimulator2/SparseSimulator2.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Common;
+using System.Runtime.InteropServices;
+using Microsoft.Quantum.Simulation.Simulators.Exceptions;
+using Microsoft.Quantum.Intrinsic.Interfaces;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.Quantum.Simulation.Simulators
+{
+    using QubitIdType = System.IntPtr;
+
+    public partial class SparseSimulator2 : CommonNativeSimulator
+    {
+        /// <summary>
+        /// Creates a an instance of a sparse simulator.
+        /// </summary>
+        /// <param name="throwOnReleasingQubitsNotInZeroState"> If set to true, the exception is thrown when trying to release qubits not in zero state. </param>
+        /// <param name="randomNumberGeneratorSeed"> Seed for the random number generator used by a simulator for measurement outcomes and the Random operation. </param>
+        /// <param name="disableBorrowing"> If true, Borrowing qubits will be disabled, and a new qubit will be allocated instead every time borrowing is requested. Performance may improve. </param>
+        /// <param name="num_qubits"> Qubit capacity. </param>
+        public SparseSimulator2(
+            bool throwOnReleasingQubitsNotInZeroState = true,
+            UInt32? randomNumberGeneratorSeed = null,
+            bool disableBorrowing = false,
+            uint num_qubits = 64)
+        : base(throwOnReleasingQubitsNotInZeroState,
+               randomNumberGeneratorSeed,
+               disableBorrowing)
+        {
+            // if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
If the sparse simulator has any native dependencies like OpenMP, it may be worth keeping this logic or extracting it into a utility method somewhere.

Reply:
I didn't look at the OpenMP topic for the Sparse Simulator yet. I will get back to it later, it is on my To Do list. 
Until then I will not mark this post as resolved.

Reply:
We found that multithreading does not improve the performance significantly and thus removed OpenMP-support to improve maintainability. The reason is that with the hashmap, the simulator is even more memory bandwidth bound than the full state simulator.